src/main/java/com/idris/CustomerController.java

```diff
@@ -7,46 +7,95 @@
         return customerService.getCustomers();
     }
 
     @PostMapping
-    @Operation(summary = "Add customer")
+    @Operation(summary = "Add a new customer")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Customer created",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Customer.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid request",
+                    content = @Content),
+            @ApiResponse(responseCode = "500", description = "Internal server error",
+                    content = @Content)
+    })
     public void addCustomer(@RequestBody NewCustomerRequest request) {
         customerService.addCustomer(request);
     }
 
     @DeleteMapping("{customerId}")
-    @Operation(summary = "Delete customer")
-    public void deleteCustomer(@PathVariable("customerId") Integer id) {
+    @Operation(summary = "Delete a customer")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Customer deleted",
+                    content = @Content),
+            @ApiResponse(responseCode = "400", description = "Invalid customer ID",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Customer not found",
+                    content = @Content)
+    })
+
+    public ResponseEntity<String> deleteCustomer(@PathVariable("customerId") Integer id) {
         customerService.deleteCustomer(id);
+        return ResponseEntity.ok("Customer with ID " + id + " was deleted successfully.");
     }
 
     @PutMapping("{customerId}")
-    @Operation(summary = "Update customer")
-    public void updateCustomer(@PathVariable("customerId") Integer id, @RequestBody NewCustomerRequest request) {
-        customerService.updateCustomer(id, request);
-    }
+    @Operation(summary = "Update an existing customer")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Customer updated",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Customer.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid request",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Customer not found",
+                    content = @Content),
+            @ApiResponse(responseCode = "500", description = "Internal server error",
+                    content = @Content)
+    })
+    public ResponseEntity<Customer> updateCustomer(@PathVariable("customerId") Integer id, @RequestBody NewCustomerRequest request) {
+        Customer updatedCustomer = customerService.updateCustomer(id, request);
+        return ResponseEntity.ok(updatedCustomer);
 }
+    @ExceptionHandler(CustomerNotFoundException.class)
```

Comment:
Does this return JSON? See https://stackoverflow.com/questions/32869714/how-to-get-response-in-json-format-using-exceptionhandler-in-spring-mvc